roles/agnosticv/templates/resource_provider.yml.j2

```diff
@@ -65,6 +65,11 @@ spec:
     - add
     - replace
     pathMatch: /spec/vars/desired_state
+{% if merged_vars.__meta__.updateFilters | default([]) | length > 0 %}
```

Comment:
@jkupferer, what would be your preference on the depth/naming convention where such a configuration item should be stored.

Is `__meta__.updateFilters` ok or would it be better placed within `.catalog`

Reply:
Not sure about the naming or where it should fit, `__meta__.anarchy` or poolboy.

But just a comment as a note for later @jkupferer : do not forget to update babylon OpenAPI schema  when this PR is merged.

here https://github.com/redhat-gpe/agnosticv/blob/master/.schemas/babylon.yaml
and 
here https://github.com/redhat-gpte/gpte_summit_2021/blob/main/.schemas/babylon.yaml

Reply:
Not sure I am entirely following correctly @jkupferer.

The existing `__meta__.catalog.parameters` are currently templated out under [job_vars](https://github.com/redhat-gpte-devopsautomation/agnosticv-operator/blob/17e6b77ebdf2417a15839bb00d03893502ffd9d7/roles/agnosticv/templates/resource_provider.yml.j2#L100) or did you mean just to use that as an example?

I've updated this branch to use `spec_parameters` in a similiar fashion to `requester_parameters` as shown below, let me know if this was incorrect.

Given this sample config,

```yaml
__meta__:
  catalog:
    spec_parameters:
    
      - name: updateFilters
        description: Custom Poolboy ResourceProvider updateFilters
        formLabel: updateFilters
        openAPIV3Schema:
          type: object
          default: {}
        updateFilters:
          - pathMatch: "/spec/vars/job_vars((?!/uuid)/.*)"
            allowedOps:
              - add
              - remove
              - replace
          - pathMatch: "/spec/hello/world"
```

Results in,

```yaml
  template:
    enable: true
  updateFilters:
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/action_schedule/start
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/action_schedule/stop
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/desired_state
  - allowedOps:
    - add
    - remove
    - replace
    pathMatch: /spec/vars/job_vars((?!/uuid)/.*)
  - pathMatch: /spec/hello/world
  validation:
    openAPIV3Schema:
      additionalProperties: false
      properties:
        spec:
          additionalProperties: false
          properties:
            updateFilters:
              default: {}
              description: Custom Poolboy ResourceProvider updateFilters
              type: object
            vars:
              additionalProperties: false
              properties:
                action_schedule:
                  additionalProperties: false
                  properties:
                    start:
                      pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
                      type: string
                    stop:
                      pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
                      type: string
                  type: object
                job_vars:
                  additionalProperties: false
                  type: object
              type: object
          required:
          - vars
          type: object
      required:
      - spec
      type: object

```

Reply:
@jkupferer, any feedback if this is what you had in-mind?

Reply:
I was thinking simpler than this. The user configuring the parameter shouldn't have to know the whole path, simply that this is a parameter than can be updated after the initial provision...

So if an item had:

```
__meta__:
  catalog:
    parameters:
      - name: example_var
        description: Sets something that can change later.
        formLabel: Example
        allowUpdate: true
        openAPIV3Schema:
          type: string
          default: ""
          pattern: ^[a-z0-9]*$
```

That would translate to:
```
  updateFilters:
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/action_schedule/start
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/action_schedule/stop
  - allowedOps:
    - add
    - replace
    pathMatch: /spec/vars/desired_state
  - allowedOps:
    - add
    - remove
    - replace
    pathMatch: /spec/vars/job_vars/example_var
```

Actually... we don't have semantics yet for an optional parameter, so `remove` would just be theoretical, but I don't see a reason to not allow it... checking the source, `allowedOps` including all three is already the default. So it could just be:

```
  updateFilters:
  ... snip ...
  - pathMatch: /spec/vars/job_vars/example_var

```

Reply:
Part of the magic here is that all parameters always set `/spec/vars/job_vars`, since these translate to variables passed into the ansible job. The user maintaining AgnosticV doesn't need to know that detail and it is probably better for future-proofing if AgnosticV doesn't reference that.

Reply:
Thanks @jkupferer, I've moved to `_meta__.catalog.parameters` but in this use case im not updating job_vars so not sure the last part makes sense.

Also, when testing using these individually generated from template I have the same requirement for setting the correct `match` as it seems like a method for this doesn't already exist?

I've updated the original comment with the current example.

Reply:
@jkupferer, any feedback on this approach?

Reply:
@jkupferer, @oybed thanks for the feedback in the session today. 

I have updated with the discussed changes and will test it out today and report back here so the PR is ready for review/merge in your AM.

Cheers.